server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -322,144 +321,150 @@ public ValuesSourceRegistry getValuesSourceRegistry() {
-                NestedAggregationBuilder::parse).addResultReader(InternalNested::new));
+                NestedAggregationBuilder::parse).addResultReader(InternalNested::new), builder);
         registerAggregation(new AggregationSpec(ReverseNestedAggregationBuilder.NAME, ReverseNestedAggregationBuilder::new,
-                ReverseNestedAggregationBuilder::parse).addResultReader(InternalReverseNested::new));
+                ReverseNestedAggregationBuilder::parse).addResultReader(InternalReverseNested::new), builder);
         registerAggregation(new AggregationSpec(TopHitsAggregationBuilder.NAME, TopHitsAggregationBuilder::new,
-                TopHitsAggregationBuilder::parse).addResultReader(InternalTopHits::new));
+                TopHitsAggregationBuilder::parse).addResultReader(InternalTopHits::new), builder);
         registerAggregation(new AggregationSpec(GeoBoundsAggregationBuilder.NAME, GeoBoundsAggregationBuilder::new,
                 GeoBoundsAggregationBuilder.PARSER)
                     .addResultReader(InternalGeoBounds::new)
-                    .setAggregatorRegistrar(GeoBoundsAggregationBuilder::registerAggregators));
+                    .setAggregatorRegistrar(GeoBoundsAggregationBuilder::registerAggregators), builder);
         registerAggregation(new AggregationSpec(GeoCentroidAggregationBuilder.NAME, GeoCentroidAggregationBuilder::new,
                 GeoCentroidAggregationBuilder.PARSER)
                     .addResultReader(InternalGeoCentroid::new)
-                    .setAggregatorRegistrar(GeoCentroidAggregationBuilder::registerAggregators));
+                    .setAggregatorRegistrar(GeoCentroidAggregationBuilder::registerAggregators), builder);
         registerAggregation(new AggregationSpec(ScriptedMetricAggregationBuilder.NAME, ScriptedMetricAggregationBuilder::new,
-                ScriptedMetricAggregationBuilder.PARSER).addResultReader(InternalScriptedMetric::new));
+                ScriptedMetricAggregationBuilder.PARSER).addResultReader(InternalScriptedMetric::new), builder);
         registerAggregation((new AggregationSpec(CompositeAggregationBuilder.NAME, CompositeAggregationBuilder::new,
-                CompositeAggregationBuilder.PARSER).addResultReader(InternalComposite::new)));
-        registerFromPlugin(plugins, SearchPlugin::getAggregations, this::registerAggregation);
+                CompositeAggregationBuilder.PARSER).addResultReader(InternalComposite::new)), builder);
+        registerFromPlugin(plugins, SearchPlugin::getAggregations, (agg) -> this.registerAggregation(agg, builder));
 
         // after aggs have been registered, see if there are any new VSTypes that need to be linked to core fields
-        registerFromPlugin(plugins, SearchPlugin::getBareAggregatorRegistrar, this::registerBareAggregatorRegistrar);
+        registerFromPlugin(plugins, SearchPlugin::getBareAggregatorRegistrar,
```

Comment:
Maybe it isn't worth having the method named `registerBareAggregatorRegistrar` now.

Reply:
Just use the lambda.